contributors/design-proposals/storage/csi-migration.md

```diff
@@ -287,8 +287,31 @@ existing Pods in the ADC.
 
 
 ### Volume Resize
+#### Offline Resizing
+In the OperationGenerator, `GenerateExpandVolumeFunc` is used to expand a volume after 
+Pod is deleted. At the beginning of the `GenerateExpandVolumeFunc`, we will check whether 
+the volume supports migration and whether migration is enabled. If enabled, we will return
+a noop function to by-pass expand controller.
 
-TODO: Design
+In the OperationGenerator, `GenerateMountVolumeFunc` is used to expand file system after volume
+is expanded and staged/mounted. The migration logic is covered by previous migration of volume mount.
+
+For external resizer, we will update it to expand volume for both CSI volume and in-tree 
+volume (only if migration is enabled). For migrated in-tree volume, it will update in-tree PV object
+with new volume size and mark in-tree PVC as resizing finished.
+
+To synchronize between expand controller and external resize, external resizer will find provisioner name
+using PVC annotation `volume.beta.kubernetes.io/storage-provisioner`. If the name matches to the CSI driver
+name, it guarantees that the PV is provisioned by CSI driver, hence external resizer will proceed with
```

Comment:
can you add to the doc why this is guaranteed, and what is the difference between native CSI usage and using CSI through Migration. Do we need to treat those cases differently?

Reply:
The difference is on which plugin backend is actually used. For in-tree volume, it will be in-tree provisioner name; for CSI volume, it will be CSI driver name. Updated doc